### PR TITLE
Implement LayoutAlignment for Core layouts

### PR DESCRIPTION
**Pull request by hartez**
Adds `LayoutAlignment` enum to Core;

Adds 
`LayoutAlignment HorizontalLayoutAlignment { get; }`
`LayoutAlignment VerticalLayoutAlignment { get; }`

to IFrameworkElement;

Handles vertical and horizontal alignment in the Vertical/Horizontal StackLayouts;

### CODE REVIEW

File: src/Core/src/Primitives/LayoutAlignment.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Microsoft.Maui.Primitives
+{
+	// We don't use Microsoft.Maui.Controls.LayoutAlignment directly because it has a Flags attribute, which we do not want
```
Comment (mattleibow) [inline src/Core/src/Primitives/LayoutAlignment.cs:3]: Is it even possible to have multiple of these? What is `Start | End` even doing?

Actually, the enum in the Controls is invalid. It is _not_ a flaggable enum:
 - `Start | Center` == `Center`
 - `Start | End` == `End`
 - `Center | End` == `Fill`

We might be better off just removing the `[Flags]` and at that point it is almost the same, just with different ordering.

File: src/Core/src/Layouts/VerticalStackLayoutManager.cs [hunk 2]
```diff
@@ -41,17 +40,16 @@ static Size Measure(double widthConstraint, int spacing, IReadOnlyList<IView> vi
...
 
-		static void Arrange(int spacing, IEnumerable<IView> views)
+		static void Arrange(double width, int spacing, IEnumerable<IView> views)
 		{
 			double stackHeight = 0;
 
 			foreach (var child in views)
 			{
-				var destination = new Rectangle(0, stackHeight, child.DesiredSize.Width, child.DesiredSize.Height);
+				var destination = new Rectangle(0, stackHeight, width, child.DesiredSize.Height);
```
Comment (mattleibow) [inline src/Core/src/Layouts/VerticalStackLayoutManager.cs:49]: Is this a "force the child to be the width" or is it just the value that is suggested to the child and the child can use a narrower width?

Comment (hartez) [reply src/Core/src/Layouts/VerticalStackLayoutManager.cs:49]: The latter. The stack layout is arranging the child within that area; the child itself determines what its final Frame is. When determining its Frame, it uses the LayoutAlignment. If the alignment is Fill, it uses all the space; if Start|End|Center, it uses its DesiredSize and sets the X,Y appropriately.

File: src/Core/src/Primitives/LayoutAlignment.cs (hunk 1 above)
Comment (hartez) [reply src/Core/src/Primitives/LayoutAlignment.cs:3]: Yes, the Controls one is weird/incorrect. But removing the FlagsAttribute would be a breaking change, and it's not on the breaking changes list. We can always obsolete the Controls version in a later release and sub in the Core version. Or we can open a separate discussion about adding it as a breaking change.

>  just with different ordering. 

This is the other reason I wanted a new enum. The enum in Controls has a default value of `Start`, but View defaults both values to `Fill` (WPF does this too). I wanted the framework default to match the enum default.

**Comment by rmarinho** [comment]
@hartez  you still going to do another pass on this right?

**Comment by hartez** [comment]
> @hartez you still going to do another pass on this right?

Should be ready to go.

**Review by rmarinho — APPROVED** [review]
It's working

**Pull request merged**